### DISCUSSION TITLE
Need to fix some issues with sveltekit

### DISCUSSION THREAD

**Discussion by NVCDevelopmentTeam** [sveltejs/kit#9858] [General]
I see, sveltekit itself has a lot of problems for visually impaired users that hinder the development of a sveltekit application, Specifically:  
- Difficult to deploy on different environments: I see, when I configure sveltekit, vercel gives an application error.
- Difficult to access data or difficult to integrate with dynamic components, floating properties.
- Difficulty integrating third-party tools like google duo or disability-specific features
- The error message on the console cannot be read because the screen reader cannot reach it
- Difficult to find the build file of the application: I tried to build with the command npm run build but can't find out which directory the build file is in.
- Add support for third-party APIs: I regret not being able to integrate advanced features like video call or voice chat, voice search to my website because it lacks APIs with libraries, I think improve.
- - Source management libraries are required: this is very important for zoning new and old code to make code management easier.
- Add the feature to customize the layout by drag and drop: Editing the layout will be easier when it can be dragged and dropped and it will generate the same code as on the ide, ie can design the interface through dragging Drop and help shape a website. Like you are working on android studio and it has three steps: drag and drop, map, write code. I found it very good.
- Use APIs or separate libraries for accessibility: In react there are these libraries.
- Can build cms: From my understanding framework is a framework so it should have full support. You know php laravel is the best framework at the moment.
- Need more audio libraries or external libraries: Have you ever created your own text to speech or built your own audio game or e-book reader or screen reader using only web technology? . This is great as chromevox is also a screen reader built on top of web technology. vs code is also built on top of web technology.
- I think, svelte needs to support building cross-platform applications like react, then it will be easy to compete: You may think, it is very normal to develop mobile or desktop applications with things like react native or Electron. . However, with svelte is something quite far away. The cross-platform expansion is the opening for future development as it will be compatible with more devices. Not to mention, web 3 is coming soon which is very important.
- Difficult to create dropdown menu or collapse button, expand menu: In react can create dropdown menu and button collapsed Toggle navigation properties but in sveltekit it doesn't work as I want. Maybe, I can't apply it because it's related to bootstrap or Angular, jQuery so I can't create collapsed button in sveltekit. Perhaps need to supplement.
- It is difficult to build a complete personal website in a short time: I think building a complete and full-featured personal website from basic to advanced is very time consuming. While, there are too many ingredients. Not to mention, embed codes or shortcodes, blocks, Widgets or Shortcodes. I also tried adding sidebar to sveltekit website but to no avail. That needs to be supplemented.
- The code is not clear: some old code from older svelte versions when uploaded to the new svelte version does not work. I think there needs to be a solution.
- Add markdown support libraries or external properties like txt2tags to make it easier to create documents or blogs.
- Need to have a concise project structure and syntax that is easier to remember.- Post display feature when building a blog with sveltekit will make it difficult to display the list of articles.
- too many things need to be changed in the svelte.config.js file making it difficult to configure sveltekit to run in many different environments such as localhost, auto, node, vercel, netlify, github page or hosting, vps , cloud server, ... Because the configuration settings are too complicated, even though the code is in many places, it can't run. Usually frontend or static website once you put the source code anywhere, it can be run without configuration, but with sveltekit, once not configured, it will be difficult to run the code. Instead of having to set up complicated configuration, why not let the code run on any platform and anywhere.
- Regarding the 500 internal server error, the syntax errors will be displayed on the browser, but even if it is displayed through the control panel, the screen reader cannot reach it. In addition, this error, if it encounters a logic error or a11y error, it only says internal 500 error, so this bug is very difficult to fix because I don't know where the wrong line is. Everything is too difficult to control.
- It is difficult to deploy data storage through a database like MongoDB, extracting data via json file like the following code:
    {"id":1,
"title_page": "Sample Title page",
"description":"Sample description Page",
"author":"admin",
"content": "Sample Content",
}
I think the things I mentioned above have a lot of problems but I believe it will be true for this topic. Thank.

**Comment by NVCDevelopmentTeam** [comment]
For sighted people, those issues are not important but for me as a blind person and an accessibility tester it greatly influenced my learning to code. You see, the syntax is not coherent or too hard to remember making it harder for me to build a layout of my own. Therefore, I believe that sveltekit will be a framework that is friendly to everyone and increases diversity and inclusion without hindering and increasing inequality in access to programming by disability.

**Comment by aloker** [comment]
First of all, thanks for your input. As someone who depends on accessibility features your feedback is especially valuable.

Personally, I think many of your points are described rather vaguely or fall outside the scope of Svelte or SvelteKit. For example, what errors do you get with Vercel? What's stopping you from creating a CMS with SvelteKit? 

One thing that I find particularly interesting is the idea of somehow transporting the server log to the client, at least in development mode. I'm not even sure how that could work reliably in practice, but it would certainly be nice to have.

The build output folder depends on the adapter. For the node adapter it defaults to "build", but it can be configured (see docs at https://kit.svelte.dev/docs/adapter-node)

Regarding your other issues - you'll have to be more specific, otherwise it's difficult for the community to help.

Hopefully we'll be able to solve your issues, because as I said I'm sure your input is very valuable.

**Comment by NVCDevelopmentTeam** [comment]
Thanks for your sharing, my friends using react do not see them complaining because they see the frontEnd code can be run anywhere. In addition, moving the code from one server to another is to avoid the case that the web fails or needs to expand hosting. I think, meeting WCAG is an important thing for svelte so I think it needs to continue to improve. I know that the default of sveltekit is auto but when I push the code to vercel it gives an error and maybe to configure it will be difficult for a blind person and especially it is a newbie to programming like me, everything too complicated. We should reduce the difficulty, reduce the cumbersome tasks and simplify those things so that it is easily accessible to everyone. I see, a sveltekit template is very complete but verbose syntax is its inadequacy, I feel like I need to find a way to make the code more compact and the syntax easy to remember like python, as long as the code can be read understand, everything is simple.

**Comment by NVCDevelopmentTeam** [comment]
Regarding the above issues, since I only speak from the experience of a blind person as an accessibility tester, I have a incomplete view but from the beginning, I had difficulty with it. I think, more blind programmers refer to this discussion to solve the problem. In addition, my wording is not good, so maybe people misunderstand. I think, you should analyze everything one by one.

**Comment by NVCDevelopmentTeam** [comment]
I'm building my personal website and I don't know how to solve complex bugs and I can't find a way to make it work in vercel. Here I can share the code if you can help me, Thanks: https://github.com/NVCDevelopmentTeam/My-website

**Comment by NVCDevelopmentTeam** [comment]
I'm wondering why svelte doesn't fully respond to WCAG and a11y because I see, properties like accesskey are missed. Have you ever built a personal website yourself as a programmer with a disability and you need to understand that the keyboard shortcuts are integrated into the website to help screen reader users operate conveniently. faster, less wasted time searching for the content you like.
